zorder/src/main/scala/org/locationtech/sfcurve/zorder/ZCurve2D.scala

```diff
@@ -62,25 +60,14 @@ class ZCurve2D(resolution: Int) extends SpaceFillingCurve2D {
   def validateY(y: Double) = math.min(math.max(ymin, y), ymax)
 
   def toRanges(xmin: Double, ymin: Double, xmax: Double, ymax: Double, hints: Option[RangeComputeHints] = None): Seq[IndexRange] = {
-    import ZCurve2D.DEFAULT_MAX_RECURSION
-
-    import scala.collection.JavaConverters._
     val colMin = mapToCol(xmin)
     val rowMin = mapToRow(ymax)
     val min = Z2(colMin, rowMin)
     val colMax = mapToCol(xmax)
     val rowMax = mapToRow(ymin)
     val max = Z2(colMax, rowMax)
 
-    val maxRecurse =
-      hints.map { h =>
-        h.asScala
-          .get(ZCurve2D.MAX_RECURSE)
-          .map { l => Try(l.asInstanceOf[Int]).getOrElse(DEFAULT_MAX_RECURSION) }
-          .getOrElse(DEFAULT_MAX_RECURSION)
-      }.getOrElse(DEFAULT_MAX_RECURSION)
-
-    Z2.zranges(min, max, maxRecurse)
+    Z2.zranges(min, max)
```

Comment:
Having a default way to compute the maximum recursion seems sensible, but why did we remove entirely the user's ability to override that?